src/components/tests/loop-form.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { createMount } from '@material-ui/core/test-utils';
+import { LoopForm } from '../loop-form';
+import React from 'react';
+
+const mount = createMount();
+
+let isRequestButtonClicked = false;
+const onClick = (
+  fleetName: string,
+  numLoops: number,
+  startLocationPoint: string,
+  endLocationPoint: string,
+  ) => {
+    isRequestButtonClicked = true;
+    console.log('test');
+  };
+
+const buildWrapper = (fleetName: string, onClick: any ) => {
+  const wrapper = mount(<LoopForm requestLoop={onClick} fleetNames={[fleetName]} />);
+  return wrapper;
+};
+
+describe('form Validation', () => {
+  afterEach(() => {
+    isRequestButtonClicked = false;
+  });
```

Comment:
Just food for thought, if in the future I create a test outside of this `describe` and if that new test changes the value `isRequestButtonClicked  =  true` and later a test inside the `describe` is executed (let's say 'Number of loops cannot be empty'), I think that 'Number of loops cannot be empty' could fail because it's expecting for `isRequestButtonClicked` to be `false`.

Thinking in that edge case, wouldn't it be better to use `beforeEach` instead of `afterEach`? 


It's also possible that I'm missing something =)

Reply:
I agree, setting the pre-conditions in `beforeEach` is better than setting it in `afterEach` for the next test. Unless there is another use for the var that I'm missing.

To improve it more, we can also move the var inside the `describe` block.

Reply:
you mean moving the initialisation of `isRequestButtonClicked` to inside the describe block?

Reply:
yea,

```js
describe('something', () => {
  let foo = false;
});
```